### PR TITLE
test: utilities for unit testing components

### PR DESCRIPTION
**Pull request by zzmp**
- Adds utilities for unit testing components
- Typechecks all tests (by moving the build tsconfig to its own `tsconfig.build.json` file)

### CODE REVIEW

File: src/cosmos/Swap.fixture.tsx [hunk 1]
```diff
@@ -8,18 +12,14 @@ import {
   SUPPORTED_LOCALES,
   SupportedChainId,
   SwapWidget,
-} from '@uniswap/widgets'
-import { useEffect } from 'react'
-import { useValue } from 'react-cosmos/fixture'
-
-import { DAI, USDC_MAINNET } from '../constants/tokens'
+} from '../index'
```
Comment (JFrankfurt) [inline src/cosmos/Swap.fixture.tsx:15]: this seems fine, but curious what the change gets us?

File: src/cosmos/Swap.fixture.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ function Fixture() {
   const [width] = useValue('width', { defaultValue: 360 })
 
   const locales = [...SUPPORTED_LOCALES, 'fa-KE (unsupported)', 'pseudo']
-  const locale = useOption('locale', { options: locales, defaultValue: DEFAULT_LOCALE, nullable: false })
+  const locale = useOption<string>('locale', { options: locales, defaultValue: DEFAULT_LOCALE, nullable: false })
```
Comment (JFrankfurt) [inline src/cosmos/Swap.fixture.tsx:44]: is locale not typed implicitly by the initialized value?

File: tsconfig.json [hunk 3]
```diff
@@ -24,10 +24,11 @@
     "skipLibCheck": true,
     "strict": true,
     "strictNullChecks": true,
+    "suppressImplicitAnyIndexErrors": true,
```
Comment (JFrankfurt) [inline tsconfig.json:27]: this feels like overkill (related to dropping the SupportedChainId?) can you just use ts-ignore where you need it instead?

File: src/cosmos/Swap.fixture.tsx (hunk 1 above)
Comment (zzmp) [reply src/cosmos/Swap.fixture.tsx:15]: `@uniswap/widgets` is self-referential, so it's outside of the type system. I can probably add it as a path, though, which might make this look a little nicer.

**Pull request merged**